editor/src/document/document_file.rs

```diff
@@ -406,7 +406,7 @@ impl MessageHandler<DocumentMessage, &InputPreprocessor> for DocumentMessageHand
 					}
 					.into(),
 				);
-				let scale = 0.5 + 0.5 * self.layerdata(&[]).scale;
+				let scale = 0.7 + 0.3 * self.layerdata(&[]).scale;
```

Comment:
Unfortunately changing 0.5 and 0.5 to 0.7 and 0.3 doesn't seem to really fix the algorithm, which is that we need to add some extra space to the ends at all times. I suggest taking the final calculation, scaling it by 0.9, and adding 0.05. That way you take whatever your calculation was and convert it into something that is 10% smaller and you will therefore always have 5% of the scrollbar track's total width on either end.

Reply:
Done